drivers/clock_management/CMakeLists.txt

```diff
@@ -0,0 +1,16 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_include_directories(.)
+zephyr_library()
+
+zephyr_library_sources(clock_management_common.c)
+zephyr_library_sources_ifdef(CONFIG_CLOCK_MANAGEMENT_FIXED_SOURCE fixed_clock_source.c)
+zephyr_library_sources_ifdef(CONFIG_CLOCK_MANAGEMENT_SOURCE clock_source.c)
+# Header for common clock drivers
+add_clock_management_header(clock_management_drivers.h)
+
+add_subdirectory(nxp_syscon)
+
+# Include headers for all clock management drivers, registered with add_clock_management_header
+zephyr_library_compile_options("-include$<JOIN:$<TARGET_PROPERTY:clock_management_header_target,SOURCES>,;-include>")
```

Comment:
Now, I appear to get this error, not quite sure what this is....
```
FAILED: zephyr/drivers/clock_management/CMakeFiles/drivers__clock_management.dir/clock_management_common.c.obj 
/home/ryanmcclelland/fbsource/arvr/third-party/toolchains/gcc-arm-none-eabi/12.2-Rel1/arm-gcc-linux/bin/arm-none-eabi-gcc -DKERNEL -DK_HEAP_MEM_POOL_SIZE=4096 -DPB_MAX_REQUIRED_FIELDS=64 -D_ANSI_SOURCE -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -Wformat -Wformat-security -Wno-format-zero-length -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/app=CMAKE_SOURCE_DIR -fmacro-prefix-map=/zephyr=ZEPHYR_BASE -ffunction-sections -fdata-sections -specs=nano.specs -D_POSIX_THREADS -std=c11 -include -MD -MT zephyr/drivers/clock_management/CMakeFiles/drivers__clock_management.dir/clock_management_common.c.obj -MF zephyr/drivers/clock_management/CMakeFiles/drivers__clock_management.dir/clock_management_common.c.obj.d -o zephyr/drivers/clock_management/CMakeFiles/drivers__clock_management.dir/clock_management_common.c.obj -c /zephyr/drivers/clock_management/clock_management_common.c
cc1: error: to generate dependencies you must specify either '-M' or '-MM'
```

Reply:
Is there an in tree sample you can modify to reproduce this? Trying to narrow it down on my end.

Reply:
alright, I found the issue it was on my end... where I also forgot to merge in the changes you made from `extenstions.cmake` as well. working good now!